std/net/curl.d

```diff
@@ -2449,6 +2449,9 @@ struct HTTP
 
                     auto m = header.findSplit(": ");
                     auto fieldName = m[0].toLower();
+                    ///Fixes https://issues.dlang.org/show_bug.cgi?id=24458
+                    if(fieldName is m[0])
+                        fieldName = fieldName.dup;
```

Comment:
should really be `idup` since the `headersIn` is `string[string]`.

Note that issue https://issues.dlang.org/show_bug.cgi?id=2954 allows a `const char[]` to be stored as the key for an immutable `string`, but that is a different bug.